docs/rules/curly/index.html

```diff
@@ -12,12 +12,47 @@
   In the code above, the author almost certainly meant for both `foo++` and `bar++`
   to be executed only if `foo === bar`. However, he forgot braces and `bar++` will be executed
   no matter what. This rule could prevent such a mistake.
-optionsDescription: Not configurable.
-options: null
+optionsDescription: |
+
+  The rule may be set to `true`, or to any combination of these arguments:
+
+  * `"all"` requires braces for all control-flow statements
+  * `"same-line-single-line"` requires braces for control-flow statements
```

Comment:
Also, sorry for not giving feedback earlier in the process. I know a lot of work went into this PR, but I don't think these are the set of options that we need.

How about this set of rules, which the user selects one of:
* always
* allow-same-line: allows for cases where the body and control flow statement are in one line. This addresses #822 

Then later, we can consider discussing:
* allow-single-line: allows for cases where the body is in a single line and may be on a different line from the control flow statement
* never-single-line
* never-same-line